src/ui/Sort/Sort.ts

```diff
@@ -142,15 +147,12 @@ export class Sort extends Component {
       this.currentCriteria = _.find(this.options.sortCriteria, (criteria: SortCriteria) => {
         return criteria.direction == direction;
       });
+      this.queryStateModel.set(QueryStateModel.attributesEnum.sort, this.currentCriteria.toString());
```

Comment:
You can extract this line and 2 other occurrences to a separate updateQueryStateModel method 

```
this.queryStateModel.set(QueryStateModel.attributesEnum.sort, this.currentCriteria.toString());
```

Reply:
Or you can move it at the end of the method (select) and remove other occurences

Reply:
Adding `updateQueryStateModel`